lib/matplotlib/axes/_axes.py

```diff
@@ -4774,6 +4775,11 @@ def fill_between(self, x, y1, y2=0, where=None, interpolate=False,
         y1 = ma.masked_invalid(self.convert_yunits(y1))
         y2 = ma.masked_invalid(self.convert_yunits(y2))
 
+        for name, array in [('x', x), ('y1', y1), ('y2', y2)]:
+            if array.ndim > 1:
+                raise ValueError('Input passed into argument "' + name +
```

Comment:
`"Input passed into argument {!r}".format(name)` (or `"... %r" % name` if you're really old-styled).
Same fixes need to be applied below.